shell/platform/darwin/macos/framework/Source/AccessibilityBridgeMacDelegate.h

```diff
@@ -0,0 +1,83 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_ACCESSIBILITYBRIDGEMACDELEGATE_H_
+#define FLUTTER_SHELL_PLATFORM_DARWIN_MACOS_FRAMEWORK_SOURCE_ACCESSIBILITYBRIDGEMACDELEGATE_H_
+
+#import <Cocoa/Cocoa.h>
+
+#include "flutter/shell/platform/common/accessibility_bridge.h"
+
+@class FlutterEngine;
+
+namespace flutter {
+
+//------------------------------------------------------------------------------
+/// The macOS implementation of AccessibilityBridge::AccessibilityBridgeDelegate.
+/// This delegate is used to create AccessibilityBridge in the macOS embedding.
+class AccessibilityBridgeMacDelegate : public AccessibilityBridge::AccessibilityBridgeDelegate {
+ public:
+  //---------------------------------------------------------------------------
+  /// @brief      Creates an AccessibilityBridgeMacDelegate.
+  /// @param[in]  flutterEngine     The weak reference to the FlutterEngine.
+  explicit AccessibilityBridgeMacDelegate(__weak FlutterEngine* flutter_engine);
+  virtual ~AccessibilityBridgeMacDelegate() = default;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  void OnAccessibilityEvent(ui::AXEventGenerator::TargetedEvent targeted_event) override;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  void DispatchAccessibilityAction(AccessibilityNodeId target,
+                                   FlutterSemanticsAction action,
+                                   const std::vector<uint8_t>& data) override;
+
+  // |AccessibilityBridge::AccessibilityBridgeDelegate|
+  std::unique_ptr<FlutterPlatformNodeDelegate> CreateFlutterPlatformNodeDelegate() override;
+
+ private:
+  /// A wrapper structure to wraps macOS native accessibility events.
+  struct NSAccessibilityEvent {
+    NSAccessibilityNotificationName name;
+    gfx::NativeViewAccessible target;
+    NSDictionary* user_info;
+  };
+
+  //---------------------------------------------------------------------------
+  /// @brief      Posts the given event against the given node to the macOS
+  ///             accessibility notification system.
+  /// @param[in]  native_node       The event target, must not be nil.
+  /// @param[in]  mac_notification  The event name, must not be nil.
+  void FireNativeMacNotification(gfx::NativeViewAccessible native_node,
+                                 NSAccessibilityNotificationName mac_notification);
```

Comment:
Here and below: maybe `DispatchMacOSNotification` for consistency with Flutter API naming, or PostMacOSNotification for consistency with macOS naming. Of the two, Dispatch is probably more in line with other engine code.